Bottom: cec7d49e6034a3d07e978524ac7e662b4b0708ce
Top:    2aad87442f67375a1d96704ceb15db2efd6f265d
Author: Andrew Cooper <andrew.cooper@novatechweb.com>
Date:   2018-07-16 15:12:24 -0500

Refresh of known_hosts

---
diff --git a/ansible-playbook/buildbot.yml b/ansible-playbook/buildbot.yml
index bc6c7663..f5bf035e 100644
--- a/ansible-playbook/buildbot.yml
+++ b/ansible-playbook/buildbot.yml
@@ -6,20 +6,20 @@
   become_user: root
   become_method: sudo
   tags:
-  - buildbot
-  pre_tasks:
+    - buildbot
+    - buildbot-master
+  tasks:
   - name: Scan Gitlab ssh keys
     command: ssh-keyscan {{gitlab_hostname}}
     register: gitlab_keys
-  roles:
   - name: Build and start the Buildbot master container
-    role: buildbot-master
-    tags:
-    - buildbot-master
+    import_role: 
+      name: buildbot-master
     vars:
       known_hosts: "{{gitlab_keys.stdout}}"
   - name: Add buildbot user to Gitlab
-    role: gitlab-user
+    import_role: 
+      name: gitlab-user
     vars:
       new_user: buildbot
       new_user_name: Build Robot
@@ -32,11 +32,11 @@
   become_user: root
   become_method: sudo
   tags:
-  - buildbot
+    - buildbot
   roles:
   - name: Add buildbot_master pubkey to buildbot user
     role: gitlab-sshkey
     vars:
       username: buildbot
-      pubkey: "{{ hostvars.buildbot_master.pubkey }}"
+      pubkey: "{{ hostvars['buildbot-master'].pubkey }}"
       title: "buildbot_master"
diff --git a/roles/buildbot-master/defaults/main.yml b/roles/buildbot-master/defaults/main.yml
index afe4bb6a..76f8b393 100644
--- a/roles/buildbot-master/defaults/main.yml
+++ b/roles/buildbot-master/defaults/main.yml
@@ -70,7 +70,7 @@ data_path: '/buildbot'
 data_volume: '{{ docker_name_prefix }}buildbot_DV'
 
 # hostname
-hostname: buildbot-worker-ntel
+hostname: '{{ role_name }}'
 
 # build arguments of the image
 image_args:
diff --git a/roles/buildbot-master/tasks/main.yml b/roles/buildbot-master/tasks/main.yml
index 3a43e6c2..4cc67d20 100644
--- a/roles/buildbot-master/tasks/main.yml
+++ b/roles/buildbot-master/tasks/main.yml
@@ -37,7 +37,7 @@
 - name: Start buildbot container
   docker_container:
     name: '{{ container_name }}'
-    hostname: '{{ buildbot_hostname }}'
+    hostname: '{{ hostname }}'
     image: '{{ image_name }}:{{ docker_image_tag }}'
     networks: '{{ container_networks }}'
     volumes: '{{ container_volumes }}'
@@ -53,6 +53,6 @@
 
 - name: Add buildbot_master host
   add_host:
-    name: buildbot_master
+    name: "{{ hostname }}"
     pubkey: "{{ output.stdout }}"
     ip_addr: "{{ buildbot_ip_addr }}"
